pkg/portal/graphql/query.go

```diff
@@ -55,5 +64,49 @@ var query = graphql.NewObject(graphql.ObjectConfig{
 				return graphqlutil.NewConnectionFromArray(out, args), nil
 			},
 		},
+		"checkCollaboratorInvitation": &graphql.Field{
+			Description: "Check whether the viewer can accept the collaboration invitation",
+			Type:        graphql.NewNonNull(checkCollaboratorInvitationPayload),
+			Args:        graphql.FieldConfigArgument{"code": &graphql.ArgumentConfig{Type:graphql.NewNonNull(graphql.String)}},
+			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				ctx := GQLContext(p.Context)
+
+				code := p.Args["code"].(string)
+
+				invitation, err := ctx.CollaboratorService.GetInvitationWithCode(code)
+				if err != nil {
```

Comment:
Use `errors.Is` or something else to distinguish the error. If the error is telling us the code is invalid, handle the error by return null. Otherwise, return the error so that we do not swallow unexpected error.